src/item.rs

```diff
@@ -1,33 +1,108 @@
+use std::collections::HashSet;
+
 use bevy::{
+    ecs::system::EntityCommands,
     math::Vec3,
-    prelude::{Bundle, Handle, Transform},
+    prelude::{Assets, Bundle, Commands, Component, Entity, Handle, Query, Transform, With},
     transform::TransformBundle,
 };
+use leafwing_input_manager::prelude::ActionState;
 
 use crate::{
-    consts::{self, ITEM_LAYER},
+    consts::{self, ITEM_LAYER, PICK_ITEM_RADIUS},
+    input::PlayerAction,
     metadata::{ItemMeta, ItemSpawnMeta},
+    player::Player,
+    state::State,
 };
 
+#[derive(Component)]
+pub struct Item;
+
+#[derive(Component)]
+pub struct CarriedBy(pub Entity);
+
+/// Represents an item, that is either on the map (waiting to be picked up), or carried.
+/// If an item is on the map, it has a TransformBundle; if it's carried, it instead has a
+/// CarriedBy.
```

Comment:
Im wondering if instead of ridding the item of its `TransformBundle` we might consider pushing it as a child of the Fighter who picks it up?

Reply:
Make sense!

Reply:
It might be awkward at first before we figure out how to integrate it with animations/what offset/rotation and whatnot it should have, but for now we could always toggle the visibility of the item off when its picked up as well.

Reply:
There you go! Done. It actually made the code a little bit simpler :smile:

Reply:
Ok. There have been several conflicts, that I had to fix. I've verified the rebase, hopefully, there have been no subtle breakages :sweat_smile:.